publisher-subscriber/README.md

```diff
@@ -0,0 +1,358 @@
+        String topicName = null;
+
+        if(args.length == 2) {
+            topicCFName = args[0];
+            topicName = args[1];
+        } else {
+            System.out.println("Invalid arguments. Should be: ");
+            System.out.println("java TLender [factory] [topic]");
+            System.exit(1);
+        }
+
+        try {
+            // Create and start activeMQ broker. Broker decouples publishers and subscribers.
+            //Additionally brokers manage threads and asynchronous sending and receiving of messages.
+            BrokerService broker = new BrokerService();
+            broker.addConnector("tcp://localhost:61616");
+            broker.start();
+
+        } catch(Exception e) {
+            e.printStackTrace();
+            LOGGER.error("An error has occurred!", e);
+        }
+
+        TLender tLender = new TLender(topicCFName, topicName);
+
+        System.out.println ("TLender Application Started");
+        System.out.println ("Press enter to quit application");
+        System.out.println ("Enter: Rate");
+        System.out.println("\ne.g. 6.8");
+
+        try {
+            BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+            //Continuously read input and send as message to subscribers
+            while(true) {
+                System.out.print("> ");
+                String line = reader.readLine();
+                //Exit if user pressed enter or line is blank
+                if (line == null || line.trim().length() == 0) {
+                    System.out.println("Exiting...");
+                    tLender.exit();
+                }
+                else { //publish the entered rate
+                    double newRate = Double.parseDouble(line);
+                    tLender.publishRate(newRate);
+                }
+            }
+        } catch(IOException e) {
+            e.printStackTrace();
+            LOGGER.error("An error has occurred!", e);
+        }
+    }
+
+    public TopicConnection gettConnection() {
+        return tConnection;
+    }
+
+    public TopicSession gettSession() {
+        return tSession;
+    }
+
+    public Topic getTopic() {
+        return topic;
+    }
+}
+```
+
+The Borrower class acts as a subscriber to a given topic, in this case mortgage interest rates. Evertime a new rate is published the subscriber receives the message.
+
+output:
+
+Initial rate is 6.0
+Waiting for new rates...
+Press enter to quit application
+
+Running the class:
+
+The class must be run after the TLender class is running since TLender spins up the activeMQ broker.
+
+In order to see the messages being sent to multiple subscribers multiple instance of the TBorrower class need to be run. Either run multiple instances in an IDE or execute the following command in a command line from the root folder after generating the target folder:
+
+
+mvn exec:java -Dexec.mainClass=com.iluwatar.publishersubscriber.TBorrower -Dexec.args="TopicCF RateTopic 6"
```

Comment:
Let's move out these running instructions to a separate file and leave README.md purely for the pattern description. Maybe we can setup USAGE.md or something similar...